api/src/main/java/org/openmrs/module/xdsbrepository/impl/XDSbServiceImpl.java

```diff
@@ -596,9 +599,49 @@ protected Location findOrCreateLocation(ExtrinsicObjectType eo) {
 			encounterLocation = locationService.saveLocation(encounterLocation);
 		}
 
+		//set software version for location
+		VersionInfoType softwareVersion = eo.getContentVersionInfo();
+		if (softwareVersion != null) {
+			encounterLocation = setSoftwareVersionForLocation(softwareVersion, encounterLocation);
+		}
+
 		return encounterLocation;
 	}
 
+	private Location setSoftwareVersionForLocation(VersionInfoType softwareVersion, Location encounterLocation) {
+		LocationAttributeType type = getSoftwareVersionAttributeType();
+		LocationAttribute attribute = null;
+
+		for (LocationAttribute tmpAttribute : encounterLocation.getAttributes()) {
+			if (tmpAttribute.getAttributeType().getUuid().equals(type.getUuid())) {
+				attribute = tmpAttribute;
```

Comment:
Do we also want to break here?

Reply:
Yes, you have right. I corrected that, thanks.